### DISCUSSION TITLE
Storybook, global styles (Kendo) and CSS modules

### DISCUSSION THREAD

**Discussion by slapadominik** [storybookjs/storybook#18754] [Help]
Hello.

I've got a problem in a Storybook with applying global CSS styles and CSS modules imported in each component. In our application we use React, NX as a build system (https://nx.dev/) and Kendo UI as a UI component library. I know from the docs that I need to import global styles in a file `.storybook/preview.js`.  That's what we did. We have bunch of UI components, each component is taken from KendoUI and we tailor it to our needs. Each component has its own CSS module imported with styles which should override global styles from Kendo. It's working perfectly in local development with hot reloading (`nx run ui-components:storybook`). However, when I build the Storybook's app (`nx build-storybook ui-components`) and then serve built bundle with webserver (`webserver 9987 ui-compmonents`)  then I encounter a problem with wrong components' styles. Components` styles from CSS modules are not overriding all of the global styles from Kendo and we have mismatch between local development and deployed version of the Storybook app. 

Below is a dynamically run application (`nx run ui-components:storybook`). You can see that Toggle's styles from `Toggle.module.scss` are overriding global styles from `node_modules/@progress/kendo-theme-default/dist/all.scss`.
![xlab1](https://user-images.githubusercontent.com/23255763/180195858-6a2f7d66-7801-4f5e-ada7-c04eff1bd770.jpg)

However after building bundle and serving it with webserver not all styles from `Toggle.module.scss` overrides global styles and there is a mix of these.
![xlab2](https://user-images.githubusercontent.com/23255763/180198413-03a8b64b-127c-472b-afdd-ec51ee74b824.jpg)

I guess that this behaviour is caused by CSS loading order in the bundle. Global styles seems to load after components` CSS modules. Is there anything that I can do to change loading order of the stylesheets? Some Webpack configuration? Or am I missing something here?

`.storybook/preview.js`
```javascript
import { ThemeContext, ThemeProvider } from '@themes';

import { GROUPS } from './../src/lib/componentsGroups';
import Badges from './badges';
import i18n from './i18nInit';
import '../../../theme/kendo-custom-theme.scss';

export const parameters = {
  options: {
    storySort: {
      method: 'alphabetical',
      order: [
        GROUPS.FORM,
        GROUPS.INPUTS,
        GROUPS.FILES,
        GROUPS.TEXT,
        GROUPS.DATA,
        GROUPS.SURFACES,
        GROUPS.LAYOUT,
        GROUPS.NAVIGATION,
        GROUPS.MODALS,
        GROUPS.FEEDBACK,
        GROUPS.OTHERS,
      ],
    },
  },
  controls: { expanded: true },
  layout: 'centered',
  badgesConfig: {
    [Badges.UNDER_DEVELOPMENT]: {
      contrast: '#B0C6CE',
      color: '#4381C1',
      title: 'UNDER DEVELOPMENT',
    },
    [Badges.BETA]: {
      contrast: '#FFAE03',
      color: '#E9190F',
      title: 'BETA / Ready for testing',
    },
    [Badges.STABLE]: {
      contrast: '#70EE9C',
      color: '#404E5C',
      title: 'STABLE',
    },
  },
};

export const globalTypes = {
  locale: {
    name: 'Locale',
    description: 'Internationalization locale',
    defaultValue: 'en',
    toolbar: {
      icon: 'globe',
      items: [
        { value: 'en', right: '🇺🇸', title: 'English' },
        { value: 'pl', right: '🇵🇱', title: 'Polski' },
      ],
    },
  },
};

export const decorators = [
  (Story, config) => {
    const {
      globals: { locale },
    } = config;

    i18n.changeLanguage(locale);

    return (
      <ThemeProvider>
        <ThemeContext.Consumer>
          {(context) => (
            <div className={context.theme}>
              <Story />
            </div>
          )}
        </ThemeContext.Consumer>
      </ThemeProvider>
    );
  },
];
```
`.storybook/main.js`
```javascript

const rootMain = require('../../../.storybook/main');

function removeUrlLoaderInSvg(config) {
  const svgRule = config.module.rules.find((rule) =>
    rule.test.toString().includes('svg')
  );
  const urlLoaderIndex = svgRule.oneOf.findIndex((rule) =>
    rule.use[0].loader.includes('url-loader')
  );
  if (urlLoaderIndex) {
    svgRule.oneOf.splice(urlLoaderIndex, 1);
  }
}

module.exports = {
  ...rootMain,

  core: { ...rootMain.core, builder: 'webpack5' },
  stories: [
    ...rootMain.stories,
    '../src/lib/**/*.stories.mdx',
    '../src/lib/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: [
    ...rootMain.addons,
    '@nrwl/react/plugins/storybook',
    '@storybook/addon-toolbars',
  ],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, { configType });
    }

    config.output.publicPath = process.env.NX_BASEPATH;

    // add your own webpack tweaks if needed
    removeUrlLoaderInSvg(config);
    return config;
  },
};
```

`Toggle.tsx`
```typescript
import { Switch, SwitchProps } from '@progress/kendo-react-inputs';
import { FC, forwardRef } from 'react';

import styles from './Toggle.module.scss';

//code omitted for brevity

const RawToggle: React.FC<ToggleProps> = forwardRef(
  (
    { label, defaultChecked, disabled, dataTestId, ...rest },
    ref: React.Ref<Switch>
  ) => (
    <div className={styles.toggle}>
      <Switch
        onLabel=""
        offLabel=""
        size="small"
        defaultChecked={defaultChecked}
        disabled={disabled}
        ref={ref}
        {...rest}
      />
      <span className={styles.label}>{label}</span>
    </div>
  )
);
```

**Comment by asotog** [comment]
@slapadominik having exact same issue, figured out global styles imported are injected with js, while CSS modules from components are bundled in a single CSS file, not sure how to fix still
